src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/OverrideMethodsOnComparableTypes.cs

```diff
@@ -133,14 +133,14 @@ private static bool IsComparableWithBaseNotComparable(INamedTypeSymbol namedType
             return true;
         }
 
-        private static bool IsComparableCore(INamedTypeSymbol namedTypeSymbol, INamedTypeSymbol comparableType, INamedTypeSymbol genericComparableType)
+        private static bool IsComparableCore(INamedTypeSymbol namedTypeSymbol, INamedTypeSymbol? comparableType, INamedTypeSymbol? genericComparableType)
             => namedTypeSymbol.AllInterfaces.Any(t => t.Equals(comparableType) ||
                                                     (t.ConstructedFrom?.Equals(genericComparableType) ?? false));
 
         private static string GetNeededComparisonOperators(INamedTypeSymbol symbol)
         {
             bool first = true;
-            StringBuilder sb = null;
+            StringBuilder? sb = null;
             void Append(string @operator)
             {
                 if (first)
```

Comment:
💭 If you just use a null check you can avoid the extra local and the suppression operator. Not sure if better.

```suggestion
                if (sb is null)
```